### DISCUSSION TITLE
Exporting POJOs for consumption in JS project / "Emit" attribute query

### DISCUSSION THREAD

**Discussion by drk-mtr** [fable-compiler/Fable#2937] [Q&A] [answered]
Hoping someone might be able to help with a very basic JS interop query.

I have the following file:

```fsharp
// user.dto.fs
module Types

type UserDto = {
    id: int
    firstName: string
    lastName: string
}
```

I want to export this and other DTOs for consumption in a JS project. So I believe I want an `index.js` file that looks like the below (I don't fully understand ESM and the differences between the various module formats etc., so I might be wrong here):

```js
import { UserDto } from '../out-prod/user.dto.js'; 

export { UserDto };
```

My attempt looks like this (it took me an embarrassingly long time to work this out):

```fsharp
// index.fs
module Index

open Fable.Core

[<Import("UserDto", "../out-prod/user.dto.js")>]
[<Emit("export { $0 }")>]
let UserDto () = jsNative

UserDto()
```

This works, but I feel like I'm probably over-complicating things. What's the correct way to do this?

**Comment by alfonsogarciacaro** [comment, marked as answer]
Hmm, this is tricky because there's no mechanism in F# to re-export types (only type aliases but these are erased in the generated code). In this case, I'd probably just write the index.js file directly in JS, unless there's a lot of other code that you want to write directly in F#. If you still need to write it in F# maybe you can use import expressions instead of attributes, which also generate a export if the member is public. Also, you can use the `${outDir}` macro so you don't need to type the output directory manually in the import. Something like this should work:

```fsharp
open Fable.Core.JsInterop

let UserDto: JsConstructor = importMember "${outDir}/user.dto.js"
```

**Reply by drk-mtr** [reply]
Thank you! I am probably over complicating by writing in F# as you say. The outDir macro is a nice touch and I hadn't realised importMemeber would re-export public members so this is really useful, cheers.
